packages/amp/src/__tests__/amp.client.test.ts

```diff
@@ -0,0 +1,33 @@
+import { createStore, InitializedStore } from "@frontity/connect";
+import clone from "clone-deep";
+import { Packages } from "../../types";
+import Amp from "../client";
+
+describe("AMP client hydration tests", () => {
+  let store: InitializedStore<Packages>;
+
+  beforeEach(() => {
+    // Initialize the store
+    store = createStore<Packages>(clone(Amp));
+
+    // We have to mock these since the `createStore` it's not the one used
+    // on the server and client.
+    store.libraries = {
+      source: {
+        parse: jest.fn(),
+        normalize: jest.fn(),
+        stringify: jest.fn(),
+      },
+      frontity: {
+        App: jest.fn(),
+      },
+    };
+  });
+
+  test("The initial App is overwritten with the Amp one", () => {
+    const previousApp = store.libraries.frontity.App;
+    store.actions.amp.beforeCSR();
+
```

Comment:
Do you think we could add another test case here?

I was thinking that we could at least also snapshot test this so that we can actually see how `App` is being overwritten. For example, we could add the snapshot test before and after running `store.actions.amp.beforeCSR()`:

```suggestion
    expect(store.libraries.frontity.App()).toMatchInlineSnapshot();
```

Reply:
You are correct, I could do better here tests wise. At least making sure the output is gonna stay the same. 👍